packages/sign-client/src/controllers/engine.ts

```diff
@@ -1058,6 +1073,30 @@ export class Engine extends IEngine {
     }, toMiliseconds(this.requestQueueDelay));
   };
 
+  // Allows for cleanup on any sent pending requests if the peer disconnects the session before responding
+  private cleanupPendingSentRequestsForTopic = ({
+    topic,
+    error,
+  }: {
+    topic: string;
+    error: ErrorResponse;
+  }) => {
+    const pendingRequests = this.client.core.history.pending;
+    if (pendingRequests.length > 0) {
+      const forSession = pendingRequests.filter(
+        (r) => r.topic === topic && r.request.method === "wc_sessionRequest",
+      );
+      if (forSession.length > 0) {
```

Comment:
Not really needed, if `filter` returned an empty array the `.forEach` is effectively a noop anyways.